### DISCUSSION TITLE
MessageConsumer.HandleBasicDeliver:: Error consuming or processing message from BUS.

### DISCUSSION THREAD

**Discussion by danammeansbear** [rabbitmq/rabbitmq-dotnet-client#1383] [Q&A]
### Describe the bug

Hello, I am getting this error when I have queues or messages processing large amounts of data records, any idea how I can resolve this? Thank you in advance! 

Row/ID: 1/193719265
App: DataLoaderService - PRD
Msg: [RELEASE] MessageConsumer.HandleBasicDeliver:: Error consuming or processing message from BUS.
Err: System.NullReferenceException: Object reference not set to an instance of an object. at EXS.Interfaces.MessageConsumer.HandleBasicDeliver(String consumerTag, UInt64 deliveryTag, Boolean redelivered, String exchange, String routingKey, IBasicProperties properties, ReadOnlyMemory`1 body) in C:codeClinical SystemsPRDEventExtractionServiceInterfacesServiceBusConsume.cs:line 273

### Reproduction steps

1. use RabbitMQ 3.12.0 with Erlang 25.3
2.send a bunch of messages over the message bus
3. wait for the error to show up
...

### Expected behavior

not expecting to get this error.

**Comment by michaelklishin** [comment]
@danammeansbear there can be any number of reasons why a consumer runs into an exception. See [server logs](https://rabbitmq.com/logging.html#log-file-location) for clues, as well as the [Heartbeats guide](https://rabbitmq.com/heartbeats.html).

We do not guess in this community.

**Comment by michaelklishin** [comment]
I don't have a good hypothesis what may cause a `System.NullReferenceException` in what seems to be a MassTransit namespace.

RabbitMQ can and will overload slower consumers if they do not [use an appropriate prefetch value](https://rabbitmq.com/confirms.html#channel-qos-prefetch) (in-flight delivery limit; covered in tutorial 2 on rabbitmq.com as well). How that can manifest itself in a given codebase will vary, in particular, most consumers, this client, frameworks such as MassTransit or NServiceBus use consumer thread pools of sorts for handling deliveries.

It's fairly common to see the need to expand it with consumers operating at peak throughput but null pointers is not usually the exception you get in that scenario.

@lukebakken anything comes to your mind?

**Reply by lukebakken** [reply]
Seems like a bug in @danammeansbear's code: `ServiceBusConsume.cs:line 273`.

@danammeansbear if you'd like to share `EXS.Interfaces.MessageConsumer.HandleBasicDeliver` maybe we can provide some ideas.
